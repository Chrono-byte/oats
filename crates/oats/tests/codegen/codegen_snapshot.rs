use anyhow::Result;
use super::common;
use common::gen_ir_for_source;

#[test]
fn snapshot_add_example_ir() -> Result<()> {
    let src = std::fs::read_to_string("../../examples/add.oats")?;
    let ir = gen_ir_for_source(&src)?;
    insta::assert_snapshot!(ir);
    Ok(())
}
