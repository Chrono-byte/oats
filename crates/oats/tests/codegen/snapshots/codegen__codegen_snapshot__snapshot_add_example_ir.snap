---
source: crates/oats/tests/codegen/codegen_snapshot.rs
assertion_line: 9
expression: ir
---
; ModuleID = 'test_module'
source_filename = "test_module"
target triple = "x86_64-redhat-linux-gnu"

declare ptr @malloc(i64)

declare i64 @strlen(ptr)

declare ptr @memcpy(ptr, ptr, i64)

declare void @free(ptr)

declare void @print_f64(double)

declare void @print_str(ptr)

declare void @sleep_ms(double)

declare ptr @str_concat(ptr, ptr)

define double @oats_main() {
entry:
  ret double 0.000000e+00
}

declare void @rc_dec(ptr)

declare void @rc_weak_dec(ptr)
